### PR TITLE
Feature/feature crud remaining

### PR DESCRIPTION
**Pull request by oricat8**

### CODE REVIEW

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/PetManagerClient.java [hunk 1]
```diff
@@ -400,15 +395,13 @@ private JSONObject buildPetJson(PetData petData) {
      * @throws IllegalArgumentException When an invalid field value is passed
      */
     private void checkCorrectType(String field, Object newValue) {
-        if ((field.equals(BIRTH) || field.equals(BREED) || field.equals(PATHOLOGIES))
-            && !(newValue instanceof String)) {
+        if ((field.equals(BIRTH) || field.equals(BREED) || (field.equals(PATHOLOGIES)
```
Comment (santidrj) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/PetManagerClient.java:398]: Codacy Issue found: [Boolean expression complexity is 4 (max allowed is 3).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5403752)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Coverage increased per file
===========================
+ usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Pet.java  2
+ usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/PetManagerClient.java  3
         

Coverage decreased per file
===========================
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/PetData.java  -1
         

Complexity increasing per file
==============================
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/KcalAverageManagerClient.java  3
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/KcalAverageData.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/KcalManagerClient.java  3
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/WeekTrainingData.java  2
- usermanagerlib/src/test/java/org/pesmypetcare/usermanagerlib/clients/FreqTrainingManagerClientTest.java  1
- usermanagerlib/src/test/java/org/pesmypetcare/usermanagerlib/clients/KcalAverageManagerClientTest.java  1
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/KcalAverage.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/FreqWashData.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Kcal.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/FreqWashManagerClient.java  3
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/FreqWash.java  2
- usermanagerlib/src/test/java/org/pesmypetcare/usermanagerlib/clients/FreqWashManagerClientTest.java  1
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/KcalData.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/WeekTraining.java  2
- usermanagerlib/src/test/java/org/pesmypetcare/usermanagerlib/clients/KcalManagerClientTest.java  1
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/WeekTrainingManagerClient.java  3
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/FreqTrainingData.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/FreqTraining.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/FreqTrainingManagerClient.java  3
- usermanagerlib/src/test/java/org/pesmypetcare/usermanagerlib/clients/WeekTrainingManagerClientTest.java  1
         

Complexity decreasing per file
==============================
+ usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/PetManagerClient.java  -3
         

Clones added
============
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/KcalAverageManagerClient.java  4
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/KcalManagerClient.java  4
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/WeightManagerClient.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/FreqWashManagerClient.java  4
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/WeekTrainingManagerClient.java  4
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/FreqTrainingManagerClient.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5403752&bid=17714238)

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/PetManagerClient.java (hunk 1 above)
Comment (santidrj) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/PetManagerClient.java:398]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5403752)

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/WeekTrainingManagerClient.java [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @param date Date the weekTraining was created
+     * @param value Value the weekTraining will have
+     * @return The response code
+     * @throws ExecutionException When the update fails
+     * @throws InterruptedException When the update is interrupted
+     */
+    public int updateWeekTrainingField(String accessToken, String owner, String petName,
+                                       DateTime date, Double value)
```
Comment (AlvarTB) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/WeekTrainingManagerClient.java:187]: Maybe you should consider not using Double. Use double instead. Double is a wrapper class. Here you don't need to use any wrapper.
What does it mean to be a wrapper? I really don't know. But I know you don't need to use it here.

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Coverage increased per file
===========================
+ usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Pet.java  2
+ usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/PetManagerClient.java  3
         

Coverage decreased per file
===========================
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/PetData.java  -1
         

Complexity increasing per file
==============================
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/FreqTrainingManagerClient.java  3
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/KcalAverageManagerClient.java  3
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/KcalAverage.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/KcalManagerClient.java  3
- usermanagerlib/src/test/java/org/pesmypetcare/usermanagerlib/clients/FreqTrainingManagerClientTest.java  1
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/FreqWash.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/WeekTrainingData.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/KcalAverageData.java  2
- usermanagerlib/src/test/java/org/pesmypetcare/usermanagerlib/clients/WeekTrainingManagerClientTest.java  1
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/FreqWashData.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/KcalData.java  2
- usermanagerlib/src/test/java/org/pesmypetcare/usermanagerlib/clients/KcalAverageManagerClientTest.java  1
- usermanagerlib/src/test/java/org/pesmypetcare/usermanagerlib/clients/FreqWashManagerClientTest.java  1
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/FreqTraining.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/FreqWashManagerClient.java  3
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/FreqTrainingData.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/WeekTrainingManagerClient.java  3
- usermanagerlib/src/test/java/org/pesmypetcare/usermanagerlib/clients/KcalManagerClientTest.java  1
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/WeekTraining.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Kcal.java  2
         

Complexity decreasing per file
==============================
+ usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/PetManagerClient.java  -3
         

Clones added
============
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/FreqTrainingManagerClient.java  4
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/KcalAverageManagerClient.java  4
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/KcalManagerClient.java  4
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/WeightManagerClient.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/FreqWashManagerClient.java  4
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/WeekTrainingManagerClient.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5403752&bid=17714238)

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/PetManagerClient.java (hunk 1 above)
Comment (santidrj) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/PetManagerClient.java:398]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5403752)

Comment (santidrj) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/PetManagerClient.java:398]: Codacy Issue found: [Boolean expression complexity is 4 (max allowed is 3).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5403752)

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/KcalManagerClient.java [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @param date Date the kcal was created
+     * @param value Value the kcal will have
+     * @return The response code
+     * @throws ExecutionException When the update fails
+     * @throws InterruptedException When the update is interrupted
+     */
+    public int updateKcalField(String accessToken, String owner, String petName, DateTime date,
+                               double value) throws ExecutionException, InterruptedException {
```
Comment (AlvarTB) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/KcalManagerClient.java:183]: :D

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Pet.java  2
         

Coverage decreased per file
===========================
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/PetData.java  -1
         

Complexity increasing per file
==============================
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/FreqWash.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/WeekTrainingManagerClient.java  3
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/KcalData.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/FreqTrainingManagerClient.java  3
- usermanagerlib/src/test/java/org/pesmypetcare/usermanagerlib/clients/KcalAverageManagerClientTest.java  1
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/KcalAverageData.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/FreqTrainingData.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/FreqTraining.java  2
- usermanagerlib/src/test/java/org/pesmypetcare/usermanagerlib/clients/KcalManagerClientTest.java  1
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/WeekTrainingData.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Kcal.java  2
- usermanagerlib/src/test/java/org/pesmypetcare/usermanagerlib/clients/FreqTrainingManagerClientTest.java  1
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/KcalAverageManagerClient.java  3
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/FreqWashData.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/WeekTraining.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/FreqWashManagerClient.java  3
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/KcalManagerClient.java  3
- usermanagerlib/src/test/java/org/pesmypetcare/usermanagerlib/clients/FreqWashManagerClientTest.java  1
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/KcalAverage.java  2
- usermanagerlib/src/test/java/org/pesmypetcare/usermanagerlib/clients/WeekTrainingManagerClientTest.java  1
         

Clones added
============
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/WeekTrainingManagerClient.java  4
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/WeightManagerClient.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/FreqTrainingManagerClient.java  4
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/KcalAverageManagerClient.java  4
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/FreqWashManagerClient.java  4
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/KcalManagerClient.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5403752&bid=17714238)

**Pull request merged**